filament/include/filament/Engine.h

```diff
@@ -187,6 +187,59 @@ class UTILS_PUBLIC Engine {
     static Engine* create(Backend backend = Backend::DEFAULT,
             Platform* platform = nullptr, void* sharedGLContext = nullptr);
 
+#if UTILS_HAS_THREADING
+
+    /**
+     * A callback used with Engine::createAsync() called once the engine is initialized and it is
+     * safe to call Engine::getEngine(token).
+     * @param user   User provided parameter given in createAsync().
+     * @param token  An opaque token used to call Engine::getEngine().
+     */
+    using CreateCallback = void(void* user, void* token);
```

Comment:
API design question: why can't we just pass the Engine* in the callback instead of a token?

Reply:
Well, so it's because I have no way to call the callback from the main thread -- which is what I wanted to do ideally.
But also because I need to run some code on the main thread after the backend is initialized but before I return Engine*.

Reply:
Note that the Java bindings will be able to invoke the user callback on main thread in the end.

How expensive is the code that needs to run on the main thread when calling getEngine()?

Reply:
So I asked Dan and they don't need java bindings for this, so I'm just thinking of not providing them.

The stuff that needs to run on the main thread is mostly creating backend objects, so it's just writing commands in the command stream. Currently it also generates the color grading LUT. More refactoring is needed if we want to do this asynchronously too.

Reply:
Sounds good. The LUT is multithreaded btw so not a big deal, but we should check again how long it takes in wall time because that will contribute to the "latency" our customers are complaining about.